internal/services/account_member/resource_test.go

```diff
@@ -76,6 +76,138 @@ func testCloudflareAccountMemberBasicConfig(resourceID, emailAddress, accountID
+					resource.TestCheckResourceAttr(name, "roles.0", "05784afa30c1afe1440e79d9351c7430"),
+				),
+			},
+			{
+				// Update role in-place (tests custom marshal logic)
+				Config: testCloudflareAccountMemberRolesConfig(rnd, fmt.Sprintf("%s@example.com", rnd), accountID, "33666b9c79b9a5273fc7344ff42f953d"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, consts.AccountIDSchemaKey, accountID),
+					resource.TestCheckResourceAttr(name, "email", fmt.Sprintf("%s@example.com", rnd)),
+					resource.TestCheckResourceAttr(name, "roles.#", "1"),
+					resource.TestCheckResourceAttr(name, "roles.0", "33666b9c79b9a5273fc7344ff42f953d"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudflareAccountMember_RolesVsPolicies(t *testing.T) {
+	// Test that using roles ignores auto-generated policies to avoid diffs
+	if os.Getenv("CLOUDFLARE_API_TOKEN") != "" {
+		t.Setenv("CLOUDFLARE_API_TOKEN", "")
+	}
+
+	rnd := utils.GenerateRandomResourceName()
+	name := "cloudflare_account_member." + rnd
+	accountID := os.Getenv("CLOUDFLARE_ACCOUNT_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.TestAccPreCheck_AccountID(t)
+			acctest.TestAccPreCheck_Credentials(t)
+		},
+		ProtoV6ProviderFactories: acctest.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testCloudflareAccountMemberRolesConfig(rnd, fmt.Sprintf("%s@example.com", rnd), accountID, "05784afa30c1afe1440e79d9351c7430"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, consts.AccountIDSchemaKey, accountID),
+					resource.TestCheckResourceAttr(name, "email", fmt.Sprintf("%s@example.com", rnd)),
+					resource.TestCheckResourceAttr(name, "roles.#", "1"),
+					resource.TestCheckResourceAttr(name, "roles.0", "05784afa30c1afe1440e79d9351c7430"),
+					// Policies should not be set when using roles
+					resource.TestCheckNoResourceAttr(name, "policies.0"),
+				),
+			},
+			{
+				// Second apply should not cause any changes (stable state)
+				Config: testCloudflareAccountMemberRolesConfig(rnd, fmt.Sprintf("%s@example.com", rnd), accountID, "05784afa30c1afe1440e79d9351c7430"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, consts.AccountIDSchemaKey, accountID),
+					resource.TestCheckResourceAttr(name, "roles.#", "1"),
+					resource.TestCheckResourceAttr(name, "roles.0", "05784afa30c1afe1440e79d9351c7430"),
+				),
+			},
+		},
+	})
+}
+
 func testCloudflareAccountMemberDirectAdd(resourceID, emailAddress, accountID string) string {
 	return acctest.LoadTestCase("cloudflareaccountmemberdirectadd.tf", resourceID, emailAddress, accountID)
 }
+
+func testCloudflareAccountMemberRolesConfig(resourceID, emailAddress, accountID, roleID string) string {
+	return acctest.LoadTestCase("cloudflareaccountmemberrolesconfig.tf", resourceID, emailAddress, accountID, roleID)
+}
+
+// func TestAccCloudflareAccountMember_Policies(t *testing.T) {
```

Comment:
we dont have resource_group id to attach to get this one work...
I can probably go find what the test user's editable false resource group id and hard code it to get it work but I am not sure if this is the best approach...

Reply:
Is there any public API that you could invoke to get that ID maybe?

Reply:
Yes, I thought it was not available in terraform but I was wrong. 
But when I started testing `cloudflare_resource_groups` it hangs forever for me

<img width="949" height="358" alt="image" src="https://github.com/user-attachments/assets/fb245d68-0c50-4404-afce-f8ef54afbb2d" />

Reply:
New tests have been added accordingly